pkg/k8shandler/util.go

```diff
@@ -37,6 +37,35 @@ func appendDefaultLabel(clusterName string, labels map[string]string) map[string
 	return labels
 }
 
+func areSelectorsSame(lhs, rhs map[string]string) bool {
+
+	if len(lhs) != len(rhs) {
+		return false
+	}
+
+	for lhsKey, lhsVal := range lhs {
+		rhsVal, ok := rhs[lhsKey]
+		if !ok || lhsVal != rhsVal {
+			return false
+		}
```

Comment:
you also need to loop through rhs and make sure all of the elements in rhs are in lhs.  I'm surprised golang doesn't have a built-in map comparison.  See this thread - https://stackoverflow.com/questions/18208394/testing-equivalence-of-maps - you might consider using reflect.DeepEqual

Reply:
Since i'm ensuring they're the same length before hand, do I need to also check in the reverse?
IIRC I had tried to just use `reflect.DeepEquals` before but it wasn't catching changes... hence writing a function :-/ 
Let me confirm that today...

Reply:
>  Since i'm ensuring they're the same length before hand, do I need to also check in the reverse?

No, so what you have is ok.  If you want to use it, that's fine.